libraries/rush-lib/src/api/RushProjectSelector.ts

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { RushConfiguration } from './RushConfiguration';
+import { RushConfigurationProject } from './RushConfigurationProject';
+import { Selection } from '../logic/Selection';
+import type { ISelectorParser } from '../logic/selectors/ISelectorParser';
+import type { ITerminal } from '@rushstack/node-core-library';
+import {
+  GitChangedProjectSelectorParser,
+  IGitSelectorParserOptions
+} from '../logic/selectors/GitChangedProjectSelectorParser';
+import { NamedProjectSelectorParser } from '../logic/selectors/NamedProjectSelectorParser';
+import { TagProjectSelectorParser } from '../logic/selectors/TagProjectSelectorParser';
+import { VersionPolicyProjectSelectorParser } from '../logic/selectors/VersionPolicyProjectSelectorParser';
+import { JsonFileSelectorParser } from '../logic/selectors/JsonFileSelectorParser';
+import { SelectorError } from '../logic/selectors/SelectorError';
+import {
+  SelectorExpression,
+  IExpressionDetailedSelector,
+  ExpressionParameter,
+  IExpressionOperatorAnd,
+  IExpressionOperatorOr,
+  IExpressionOperatorNot,
+  isDetailedSelector,
+  isParameter,
+  isAnd,
+  isOr,
+  isNot
+} from './SelectorExpressions';
+
+/**
+ * When preparing to select projects in a Rush monorepo, some selector scopes
+ * require additional configuration in order to control their behavior. This
+ * options interface allows the caller to provide these properties.
+ */
+export interface IProjectSelectionOptions {
+  /**
+   * Options required for configuring the git selector scope.
+   */
+  gitSelectorParserOptions: IGitSelectorParserOptions;
+}
+
+/**
+ * A central interface for selecting a subset of Rush projects from a given monorepo,
+ * using standardized selector expressions. Note that the types of selectors available
+ * in a monorepo may be influenced in the future by plugins, so project selection
+ * is always done in the context of a particular Rush configuration.
+ */
+export class RushProjectSelector {
+  private _rushConfig: RushConfiguration;
+  private _scopes: Map<string, ISelectorParser<RushConfigurationProject>> = new Map();
+  private _options: IProjectSelectionOptions;
+
+  public constructor(rushConfig: RushConfiguration, options: IProjectSelectionOptions) {
+    this._rushConfig = rushConfig;
+    this._options = options;
+
+    this._scopes.set('name', new NamedProjectSelectorParser(this._rushConfig));
+    this._scopes.set(
+      'git',
+      new GitChangedProjectSelectorParser(this._rushConfig, this._options.gitSelectorParserOptions)
+    );
+    this._scopes.set('tag', new TagProjectSelectorParser(this._rushConfig));
+    this._scopes.set('version-policy', new VersionPolicyProjectSelectorParser(this._rushConfig));
+    this._scopes.set('json', new JsonFileSelectorParser(this._rushConfig, this));
```

Comment:
```suggestion
    const scopes: Map<string, ISelectorParser<RushConfigurationProject>> = new Map();
    this._scopes = scopes;
    scopes.set('name', new NamedProjectSelectorParser(rushConfig));
    scopes.set(
      'git',
      new GitChangedProjectSelectorParser(rushConfig, options.gitSelectorParserOptions)
    );
    scopes.set('tag', new TagProjectSelectorParser(rushConfig));
    scopes.set('version-policy', new VersionPolicyProjectSelectorParser(rushConfig));
    scopes.set('json', new JsonFileSelectorParser(rushConfig, this));
```
Reduction in verbosity, removes unnecessary property accesses, helps the code compress better.